waveshare-oled/waveshareoled-actor/src/lib.rs

```diff
@@ -1,24 +1,30 @@
 use serde_json::json;
 use wasmbus_rpc::actor::prelude::*;
 use wasmcloud_interface_httpserver::{HttpRequest, HttpResponse, HttpServer, HttpServerReceiver};
-use waveshareoled_interface::{Waveshareoled, WaveshareoledSender, DrawMessageInput};
+use waveshareoled_interface::{
+    DrawMessageInput, Event, WaveshareSubscriber, WaveshareSubscriberReceiver, Waveshareoled,
+    WaveshareoledSender,
+};
 
 #[derive(Debug, Default, Actor, HealthResponder)]
-#[services(Actor, HttpServer)]
-struct WaveshareoledActorActor {}
+#[services(Actor, HttpServer, WaveshareSubscriber)]
+struct WaveshareoledActor {}
 
 /// Implementation of HttpServer trait methods
 #[async_trait]
-impl HttpServer for WaveshareoledActorActor {
+impl HttpServer for WaveshareoledActor {
     async fn handle_request(&self, ctx: &Context, req: &HttpRequest) -> RpcResult<HttpResponse> {
-        let resp = WaveshareoledSender::new().draw_message(
-            ctx,
-            &DrawMessageInput {
-                message: req.path.clone(),
-            },
-        ).await;
-
-        let (body, status_code) = match resp {
+        let censor_filter = censor::Censor::Standard + censor::Censor::Sex;
+        let name = censor_filter.censor(req.path.trim_start_matches('/'));
+        let (body, status_code) = match WaveshareoledSender::new()
+            .draw_message(
+                ctx,
+                &DrawMessageInput {
+                    message: format!("Hello, {name}, I am Roman!",),
```

Comment:
Need to find a way to make this generic

Since actors are stateless we should probably have the provider track identity via a link def, and inject this context on incoming messages from the actor